### PR TITLE
Propagate response to the client for insertAllTableData operation

### PR DESCRIPTION
**Pull request by Biruntha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: bigquery2/bigquery_types.bal [hunk 1]
```diff
@@ -87,3 +87,20 @@ public type Table record {
...
+    int index = 0;
+    Error[] errors = [];
+};
+
+public type Error record {
+    string reason = "";
+    string location = "";
+    string debugInfo = "";
+    string message = "";
+};
```
Comment (anupama-pathirage) [inline bigquery2/bigquery_types.bal]: Can't we use ballerina builtin error here?

File: bigquery2/bigquery_types.bal [hunk 2]
```diff
@@ -87,3 +87,20 @@ public type Table record {
...
     Field[] fields?;
 };
+
+public type InsertTableData record {
+    string kind = "";
+    InsertError[] insertErrors = [];
+};
+
+public type InsertError record {
+    int index = 0;
```
Comment (anupama-pathirage) [inline bigquery2/bigquery_types.bal]: Do we need to have index explicitly here? Isn't it same as the array index of InsertError[] insertErrors?

Comment (Biruntha) [reply bigquery2/bigquery_types.bal]: I think we need to have. Here Index parameter points to the index of the row that error applies to.

https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/insertAll

File: bigquery2/bigquery_types.bal (hunk 1 above)
Comment (Biruntha) [reply bigquery2/bigquery_types.bal]: From the API we will get response like below for 200 OK. 
{
  "kind": "bigquery#tableDataInsertAllResponse",
  "insertErrors": [
    {
      "index": unsigned integer,
      "errors": [
        {
          "reason": string,
          "location": string,
          "debugInfo": string,
          "message": string
        }
      ]
    }
  ]
}
If we are using ballerina builtin error, Don't we need to pass these details "reason", "location", "debugInfo"  to user?

Comment (hasithaa) [reply bigquery2/bigquery_types.bal]: @anupama-pathirage @Biruntha , I also think using error value here is a good thing. But I am not sure current compiler supports this feature OOB. So could you please create an issue to track this improvement, (Checking Possibility of using error values to represent client errors) in Ballerinalang repo? Until then shall we continue with current approach.

Comment (Biruntha) [reply bigquery2/bigquery_types.bal]: Created the issue to track

https://github.com/ballerina-platform/ballerina-lang/issues/14297

**Pull request merged**